library/core/src/convert/mod.rs

```diff
@@ -532,7 +532,8 @@ where
 
 // From implies Into
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T, U> Into<U> for T
+#[rustc_const_unstable(feature = "const_convert", issue = "none")]
+impl<T, U> const Into<U> for T
 where
     U: From<T>,
```

Comment:
You need

```suggestion
    U: ~const From<T>,
```

now

Reply:
which needs bootstrap shenanigans, so you need to duplicate the impl and cfg it

Reply:
Ok, thanks :)

---

I tried with the following (and putting `#[cfg(bootstrap)]` on the existing impl obove)

```rust
#[cfg(not(bootstrap))]
#[stable(feature = "rust1", since = "1.0.0")]
#[rustc_const_unstable(feature = "const_convert", issue = "none")]
impl<T, U> const Into<U> for T
where
    U: ~const From<T>,
{
    fn into(self) -> U {
        U::from(self)
    }
}
```

however the compiler complains about unexpected `~`. Does that mean that the current bootstrapping compiler cannot parse `~const` even when in inactive cfg? So this needs to wait for next bootstrapping bump?

Or am I missing something :)

Reply:
oof yea, that's the issue

Reply:
No problem, I will wait with that part for another PR then :)

Reply:
The current bootstrap compiler supports `~const` syntax now.

Reply:
Thanks :)